pkg/host/host.go

```diff
@@ -0,0 +1,126 @@
+/*
+2024 NVIDIA CORPORATION & AFFILIATES
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package host
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/Mellanox/nic-configuration-operator/api/v1alpha1"
+	"github.com/Mellanox/nic-configuration-operator/pkg/consts"
+)
+import "github.com/jaypipes/ghw"
+
+// HostManager contains logic for managing NIC devices on the host
+type HostManager interface {
+	DiscoverNicDevices() map[string]v1alpha1.NicDeviceStatus
+	UpdateDeviceConditionsStatus(device *v1alpha1.NicDevice)
+}
+
+type hostManager struct {
+	nodeName  string
+	hostUtils HostUtils
+}
+
+// DiscoverNicDevices uses host utils to discover Nvidia NIC devices on the host and returns back a map of serial numbers to device statuses
+func (h hostManager) DiscoverNicDevices() map[string]v1alpha1.NicDeviceStatus {
+	log.Log.Info("HostManager.DiscoverNicDevices()")
+
+	unchroot, err := h.hostUtils.Chroot("/host")
```

Comment:
We are executing a lot of system tools, where it's not always entirely clear, which system directories they try to access. If needed, we can try to make do with a handful of specific directories mounts but that's subject to problems in the future IMHO

Reply:
If I understand correctly, this will case that we will call `mstconf` and `mstvpd` from the host (meaning these tools should be instaleld on the host OS) . I think this is not what we want.

Reply:
@ykulazhenkov Even if we install those tools inside the container, they will still need to access different system directories, and we don't know fully which ones

Reply:
this should be addressed on a ad-hoc basis (if additional mounts needed)
for now we should try to have those tool part of the container image invoking them from the container,

e,g for mstflint related tools we do so today in sriov network config daemon

Reply:
Now we install mstflint to our docker image, and only mount /sys dir (strace on all the used commands didn't show any additional dirs that we might need to mount) so far. That said, using the centos image as a base now